src/main/java/it/uniroma3/mat/extendedset/intset/ImmutableConciseSet.java

```diff
@@ -694,19 +687,26 @@ public static ImmutableConciseSet doIntersection(Iterator<ImmutableConciseSet> s
     return new ImmutableConciseSet(IntBuffer.wrap(retVal.toArray()));
   }
 
-  private static void sort(final WordIterator[] a, final int to, final Comparator<WordIterator> comp)
+  /**
+   * Variation of insertion sort, elements [maxChangedIndex + 1, size) are sorted, elements [0, maxChangedIndex] should
+   * be inserted into that sorted range.
+   */
+  private static void partialSort(
```

Comment:
I tested current version against `System.arraycopy()`:
```
Benchmark                                     (bitmapAlgo)  (n)  (prob)   (size)  Mode  Cnt          Score         Error  Units
BitmapIterationBenchmark.intersectionAndIter       concise    2     0.5  3000000  avgt    5   24454733.222 ±  511862.204  ns/op
BitmapIterationBenchmark.intersectionAndIter       concise   10     0.5  3000000  avgt    5   35222759.993 ± 1296917.013  ns/op
BitmapIterationBenchmark.intersectionAndIter       concise   30     0.5  3000000  avgt    5  133591286.100 ± 4871974.001  ns/op

Benchmark                                     (bitmapAlgo)  (n)  (prob)   (size)  Mode  Cnt          Score         Error  Units
BitmapIterationBenchmark.intersectionAndIter       concise    2     0.5  3000000  avgt    5   26250153.600 ± 1890538.052  ns/op
BitmapIterationBenchmark.intersectionAndIter       concise   10     0.5  3000000  avgt    5   37682895.481 ±  696059.156  ns/op
BitmapIterationBenchmark.intersectionAndIter       concise   30     0.5  3000000  avgt    5  142231612.521 ± 4774314.341  ns/op
```

`System.arraycopy()` is slower on sizes which we are interested in.